maven-java/kalix-maven-plugin/src/main/java/kalix/DeployMojo.java

```diff
@@ -36,30 +35,85 @@ public class DeployMojo extends AbstractMojo {
     @Parameter(property = "kalixContext")
     private String kalixContext;
 
+    @Parameter(property = "kalixCurrentProject", required = true)
+    private String kalixCurrentProject;
+
+    @Parameter(defaultValue = "false", property = "kalixDeploymentOn", required = true)
+    private Boolean kalixDeploymentOn;
+
     @Parameter(defaultValue = "30000", property = "cliTimeoutMs", required = true)
     private Long cliTimeoutMs;
 
     private final Log log = getLog();
 
     /**
-     * We deploy by invoking the services deploy command
+     * We deploy by invoking the services deploy command only when `kalixDeploymentOn` is `true`
+     * and the current Kalix project matches with 'kalixCurrentProject'.
      */
     public void execute() throws MojoExecutionException {
-        log.info("Deploying project to Kalix");
+       final List<String> commandLine;
+       final int deploymentResult = 0;
+       if (!Boolean.valueOf(kalixDeploymentOn)){
+           log.info("The variable `kalixDeploymentOn` hasn't been set to `true`. Therefore, not deploying to Kalix");
+           log.info("To deploy to Kalix via `mvn deploy` you need to set `kalixDeploymentOn` to `true` " +
+                   "and `kalixCurrentProject` to your current Kalix project (`kalix config get-project`)");
+           return;
+       }
+       if (!currentKalixProjectMatches()){
+           return;
+       }
+       if (kalixContext != null) {
+           commandLine = Arrays.asList(kalixPath, "--context", kalixContext, "service", "deploy", service, dockerImage);
+           deploy(commandLine);
+           } else {
+           commandLine = Arrays.asList(kalixPath, "service", "deploy", service, dockerImage);
+           deploy(commandLine);
+       }
+    }
+
+    private boolean currentKalixProjectMatches() throws MojoExecutionException {
+      try {
+          List<String> commandLine = Arrays.asList(kalixPath, "config", "get-project");
+          Process process = new ProcessBuilder().command(commandLine).start();
+          synchronized (process) {
+            process.wait(cliTimeoutMs);
+          }
+          final int getProjectResult = process.exitValue();
+          if (getProjectResult == 0){
+              InputStream inputStream = process.getInputStream();
+              Scanner scanner = new Scanner(inputStream).useDelimiter("\\A");
+              String currentProjectInfo = "";
+              while(scanner.hasNext()){
+                  currentProjectInfo += scanner.next();
+              }
+              if(!currentProjectInfo.contains(kalixCurrentProject) || kalixCurrentProject.isEmpty()){
+                  log.info("Your current Kalix project ID is not [" + kalixCurrentProject + "]. Therefore, not deploying to Kalix");
+                  log.info("Your current Kalix project (`kalix config get-project`) is:");
```

Comment:
It may be confusing to show a command here. 

```suggestion
                  log.info("Your current Kalix project is:");
```

Reply:
I thought this would save the newbies a few precious seconds. But I agree it doesn't look great.